File: LeetCode/1. Two Sum/Solution.cs
public class Solution 
{
    public int[] TwoSum(int[] nums, int target) 
    {
        Dictionary<int, int> indices = new Dictionary<int, int>();
        
        for(int i = 0; i < nums.Length; i++)
        {
            if(indices.ContainsKey(target-nums[i])) 
            {
            	return new int[2]{indices[target-nums[i]], i};
            }
            else 
            {
            	indices.Add(nums[i], i);
            }
        }
        
        return null;
    }
}

File: LeetCode/1. Two Sum/Solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> answer;
        
        unordered_map<int,int> seen;
        for(int i = 0; i < nums.size(); i++){
            if(seen.find(target-nums[i]) != seen.end()){
                answer.push_back(seen[target-nums[i]]);
                answer.push_back(i);
                return answer;
            }
            
            seen[nums[i]] = i;
        }
        
        return answer;
    }
};